### PR TITLE
fix: migrate consumer panic

### LINKED ISSUE #21: bug: migrate consumer panic
image

---

Use: https://github.com/Kong/deck/blob/551b414568be603d22b040b92b7a29827878feb7/kong.yaml source config.
Go Version: `go 1.16.5`
Reason: The KeyAuthCredentials array size is not checked.

### PR DESCRIPTION
**Pull request by shuaijinchao**
FIX #21

### CODE REVIEW

File: pkg/kong/consumer.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ func MigrateConsumer(kongConfig *KongConfig, configYamlAll *[]utils.YamlItem) (*
 		}
 
 		// TODO: need to test then it got multiple key
-		if c.KeyAuthCredentials[0].Key != "" {
+		if len(c.KeyAuthCredentials) > 0 && c.KeyAuthCredentials[0].Key != "" {
```
Comment (spacewander) [inline pkg/kong/consumer.go:22]: Need a test for it

File: pkg/kong/consumer_test.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package kong
+
+import (
+	"github.com/api7/kong-to-apisix/pkg/utils"
+	"github.com/stretchr/testify/assert"
+	"testing"
```
Comment (spacewander) [inline pkg/kong/consumer_test.go:6]: Please use goimport to sort it like other files, see https://github.com/api7/kong-to-apisix/blob/2c941302103026edb508cedea10c7985c54849cd/pkg/utils/utils.go#L4

File: Makefile [hunk 3]
```diff
@@ -5,3 +5,7 @@ test:
 .PHONY: build
 build:
 	go build -o bin/kong-to-apisix ./cmd/kong-to-apisix/main.go
+
+.PHONY: unit-test
+unit-test:
+	go test --count=1 ./pkg/kong
```
Comment (spacewander) [inline Makefile:11]:
```suggestion
	go test -race --count=1 ./pkg/...
```

**Comment by shuaijinchao** [comment]
@spacewander please review

File: Makefile [hunk 4]
```diff
@@ -5,3 +5,7 @@ test:
 .PHONY: build
 build:
 	go build -o bin/kong-to-apisix ./cmd/kong-to-apisix/main.go
+
+.PHONY: unit-test
+unit-test:
+	go test -race --count=1 ./pkg/kong
```
Comment (spacewander) [inline Makefile:11]:
```suggestion
	go test -race --count=1 ./pkg/...
```

Comment (shuaijinchao) [reply Makefile]: ok done

**Review by spacewander — APPROVED** [review]

**Pull request merged**